src/converse-register.js

```diff
@@ -241,6 +245,25 @@
                     return false;
                 },
 
+                chooseProvider: function (ev) {
+                    /* Method called when registration_domain is provided
+                     * which loads the registration form directly
+                     */
+                    var $form = $('form#converse-register');
+                            $form.find('input[name=domain]').hide();
```

Comment:
1. Fix the indentation please.

2. Instead of manually hiding the `input` elements, you can rather just replace the contents of the entire form with the template.

3. I'm in the process of removing jQuery from converse.js. I know there's still lots of jQuery code, but in we should at least not add any more jQuery stuff. Instead of using the jQuery API here, please use instead the native DOM API. This might help: http://youmightnotneedjquery.com/
Instead of using `.hide()`, you can use `el.classList.add('hidden')`.